packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/BaseViewManager.java

```diff
@@ -164,14 +166,26 @@ public void onLayoutChange(
   }
 
   @Override
-  @ReactProp(
-      name = ViewProps.BACKGROUND_COLOR,
-      defaultInt = Color.TRANSPARENT,
-      customType = "Color")
   public void setBackgroundColor(@NonNull T view, int backgroundColor) {
     view.setBackgroundColor(backgroundColor);
   }
 
+  @ReactProp(
+    name = ViewProps.BACKGROUND_COLOR,
+    defaultInt = Color.TRANSPARENT,
+    customType = "Color")
+  public void setBackgroundColor(@NonNull T view, long backgroundColor) {
+    try {
+      Method setBackgroundColorMethod = view.getClass().getMethod("setBackgroundColor", long.class);
+      setBackgroundColorMethod.invoke(view, backgroundColor);
```

Comment:
Can we instead use something like a `ColorDrawable` here or avoid using reflection if possible

Reply:
@cortinico I tried making `ColorDrawable` work but I don't think I can make it work with long colors. Instead I think we can just check if the view is an instance of `ReactViewGroup` (Is this always true?) and that seems to work.

Reply:
> we can just check if the view is an instance of ReactViewGroup (Is this always true?)

I don't think it's always true sadly